### PR TITLE
Allow skipping email addresses in signup

### PR DESCRIPTION
**Pull request by pzduniak**

### CODE REVIEW

File: shared/people/index.shared.tsx [hunk 1]
```diff
@@ -67,15 +68,43 @@ const EmailVerificationBanner = ({email}) => {
       () => dispatch(SignupGen.createClearJustSignedUpEmail()),
     []
   )
+  if (email === 'none') {
```
Comment (buoyad) [inline shared/people/index.shared.tsx]: Let's make this `'NOEMAIL'` and put it in constants/signup

File: shared/people/index.shared.tsx [hunk 2]
```diff
@@ -67,15 +68,43 @@ const EmailVerificationBanner = ({email}) => {
...
+    return <Kb.Banner color="green">{`Welcome to Keybase!`}</Kb.Banner>
```
Comment (buoyad) [inline shared/people/index.shared.tsx]: No need to bracket this string

File: shared/people/index.shared.tsx [hunk 3]
```diff
@@ -67,15 +68,43 @@ const EmailVerificationBanner = ({email}) => {
...
+      />
+    </Kb.Banner>
+  )
+}
+
 export const PeoplePageList = (props: Props) => (
   <Kb.Box style={{...Styles.globalStyles.flexBoxColumn, position: 'relative', width: '100%'}}>
     {Styles.isMobile && <AirdropBanner showSystemButtons={false} />}
     {!!props.signupEmail && <EmailVerificationBanner email={props.signupEmail} />}
+    {!!props.resentEmail && <ResentEmailVerificationBanner email={props.resentEmail} />}
```
Comment (buoyad) [inline shared/people/index.shared.tsx]: Can `ResentEmailVerificationBanner` get `resentEmail` out of the store and return null if not set?

File: shared/people/todo/container.tsx [hunk 4]
```diff
@@ -281,15 +282,15 @@ const VerifyAllEmailConnector = connect(
         ...(meta
           ? [
               {
-                label: hasRecentVerifyEmail ? `Resend the verification email` : 'Verify',
+                label: hasRecentVerifyEmail ? `Send the verification email again` : 'Verify',
```
Comment (buoyad) [inline shared/people/todo/container.tsx]: This is very verbose. Can we say `Resend verification` or `Resend`? Spills out of the button on iPhone 5S, it likely gets cut off mid-word on other screen sizes.
image

File: shared/settings/advanced/container.tsx [hunk 5]
```diff
@@ -53,6 +53,7 @@ const mapDispatchToProps = dispatch => ({
   onDisableCertPinning: () =>
     dispatch(RouteTreeGen.createNavigateAppend({path: ['disableCertPinningModal']})),
   onEnableCertPinning: () => dispatch(createCertificatePinningToggled({toggled: false})),
+  onEnterPhoneFlow: () => dispatch(RouteTreeGen.createNavigateAppend({path: ['signupEnterPhoneNumber']})),
```
Comment (buoyad) [inline shared/settings/advanced/container.tsx]: - [ ] remove before merge

File: shared/signup/email/container.tsx [hunk 6]
```diff
@@ -1,25 +1,67 @@
...
+import EnterEmail, {Props} from '.'
 
 type OwnProps = {}
 
+type WatcherProps = Props & {
+  addedEmail: string | null
+  onSuccess: (email: string) => void
+}
+const WatchForSuccess = (props: WatcherProps) => {
+  const [addEmailInProgress, onAddEmailInProgress] = React.useState('')
```
Comment (buoyad) [inline shared/signup/email/container.tsx]: `onAddEmailInProgress` -> `setAddEmailInProgress`

File: shared/signup/email/index.tsx [hunk 7]
```diff
@@ -1,100 +1,104 @@
...
+          onClick: onContinue,
+          type: 'Success',
+          waiting: props.waiting,
+        },
+        ...(!Styles.isMobile && props.onSkip
+          ? [
+              {
+                label: 'Skip for now',
+                onClick: props.onSkip,
+                type: 'Dim' as ButtonType,
```
Comment (buoyad) [inline shared/signup/email/index.tsx]: `as const`

File: shared/signup/email/index.tsx [hunk 8]
```diff
@@ -1,100 +1,104 @@
...
+import {SignupScreen, errorBanner} from '../common'
+import {ButtonType} from '../../common-adapters/button'
 
-type Props = {
-  error: string
-  initialEmail?: string
-  onBack?: () => void
-  onFinish: (email: string, allowSearch: boolean) => void
+export type Props = {
+  error: Error | null
```
Comment (buoyad) [inline shared/signup/email/index.tsx]: Convert this to a string in the container

File: shared/signup/email/index.tsx [hunk 9]
```diff
@@ -1,100 +1,104 @@
...
+          type: 'Success',
+          waiting: props.waiting,
+        },
+        ...(!Styles.isMobile && props.onSkip
+          ? [
+              {
+                label: 'Skip for now',
+                onClick: props.onSkip,
+                type: 'Dim' as ButtonType,
+                waiting: props.waiting,
```
Comment (buoyad) [inline shared/signup/email/index.tsx]: `disabled` rather than `waiting`

File: shared/people/todo/container.tsx (hunk 4 above)
Comment (jakob223) [reply shared/people/todo/container.tsx]: I thought the non-resend case used to just be "verify" - that seems simplest to me and also doesn't flow off the screen.

Comment (pzduniak) [reply shared/people/todo/container.tsx]: "again" is missing

Comment (pzduniak) [reply shared/people/todo/container.tsx]: switching this to "Verify again"

Comment (jakob223) [reply shared/people/todo/container.tsx]: oh I missed the again. Maybe "Retry verification"? "verify again" sounds to me like you've already verified it successfully once.

File: shared/people/container.tsx [hunk 10]
```diff
@@ -101,7 +98,6 @@ export default Container.connect(
     myUsername: stateProps.myUsername,
     newItems: stateProps.newItems,
     oldItems: stateProps.oldItems,
-    resentEmail: stateProps.resentEmail,
     showAirdrop: stateProps.showAirdrop,
     signupEmail: stateProps.signupEmail,
```
Comment (buoyad) [inline shared/people/container.tsx]: This isn't needed either right?

File: shared/signup/email/index.tsx [hunk 11]
```diff
@@ -20,22 +19,21 @@ const EnterEmail = (props: Props) => {
 
   return (
     <SignupScreen
-      banners={errorBanner(props.error ? props.error.message : '')}
+      banners={errorBanner(props.error)}
       buttons={[
         {
-          disabled,
+          disabled: disabled || props.waiting,
```
Comment (buoyad) [inline shared/signup/email/index.tsx]: Hm, but I think this one should get `waiting`. You're waiting for this button's action to finish. Whereas we want to disable canceling while this is in flight.

**Pull request merged**